### DISCUSSION TITLE
Cases of errors when segmenting and their way of debugging

### DISCUSSION THREAD

**Discussion by zstanecic** [RHVoice/RHVoice#461] [Q&A]
Hi to all again!
I think, that i am not bothering with too much questions here, but it is worth having this information here, :).
Today's topic is: types of errors in segmentation and their way of fixing. 
How the following error can be fixed, and what can be the possible problem?
hmm 12                                                                                                     
  ERROR [+6510]  LOpen: Unable to open label file htk/mfcc/RHVoice_Natan_0247.lab                                       
 FATAL ERROR - Terminating program /home/zstanecic/tts/htk341/bin/HERest                                                
Traceback (most recent call last):                                                                                      
  File "/home/zstanecic/tts/natan/../RHVoice/src/scripts/general/voice-building-utils", line 1592, in <module>          
    args.func(args)                                                                                                     
  File "/home/zstanecic/tts/natan/../RHVoice/src/scripts/general/voice-building-utils", line 601, in __call__           
    n=self.reest_until(n,2,0.005,max_reest_iter)                                                                        
  File "/home/zstanecic/tts/natan/../RHVoice/src/scripts/general/voice-building-utils", line 458, in reest_until        
    self.reest(i,j)                                                                                                     
  File "/home/zstanecic/tts/natan/../RHVoice/src/scripts/general/voice-building-utils", line 452, in reest              
    self.hrun("HERest","-C",self.param_conf_path,"-I",phmlf,"-t","250.0","150.0","1000.0","-S",self.mfcc_scp_path,"-H",os.path.join(indir,"macros"),"-H",os.path.join(indir,"hmmdefs"),"-M",outdir,phlist,tag=str(i))                           
  File "/home/zstanecic/tts/natan/../RHVoice/src/scripts/general/voice-building-utils", line 339, in hrun               
    subprocess.check_call(cmd,stdout=f)                                                                                 
  File "/usr/lib/python3.9/subprocess.py", line 373, in check_call                                                      
    raise CalledProcessError(retcode, cmd)                                                                              
subprocess.CalledProcessError: Command '['/home/zstanecic/tts/htk341/bin/HERest', '-D', '-A', '-T', '1', '-C', 'htk/param.conf', '-I', 'htk/phones2.mlf', '-t', '250.0', '150.0', '1000.0', '-S', 'htk/mfcc.scp', '-H', 'htk/hmm/12/macros', '-H', 'htk/hmm/12/hmmdefs', '-M', 'htk/hmm/13', 'htk/phonelist']' returned non-zero exit status 110.
The problem is, that even error is not so well understandable, thus making it impossible to debug.

**Comment by Olga-Yakovleva** [comment]
Check if you have the following file: htk/problematic_files_2.txt

It should list the original names of your recordings with a number 
beside each one. The higher the number, the more likely it is that the 
text doesn't match the audio in some way: a missing word, an extra word, 
a major pronunciation difference.

Check your text and audio very carefully and fix the text to match the 
audio if you see the problem.

Then run the segmentation again after you have fixed all the mismatches.
